### ISSUE TITLE
radxa-zero3_debian_bullseye_cli_b3.img doesn't boot on Zero 3W

### ISSUE THREAD

**Issue by zehnm** [radxa-build/radxa-zero3#4]
The Debian CLI image `radxa-zero3_debian_bullseye_cli_b3.img` doesn't work on a Zero 3W. Kernel hangs after `rockchip-dmc dmc: could not find power_model node` log message.

Flashed on an SD card:
```
Starting kernel ...

[   12.088516] fiq_debugger fiq_debugger.0: IRQ fiq not found
[   12.088580] fiq_debugger fiq_debugger.0: IRQ wakeup not found
[   12.088613] fiq_debugger_probe: could not install nmi irq handler
[   13.562615] mpp-iep2 fdef0000.iep: allocate roi buffer failed
[   13.582867] arm-scmi firmware:scmi: Failed. SCMI protocol 22 not active.
[   13.605075] dwhdmi-rockchip fe0a0000.hdmi: IRQ index 1 not found
[   13.864727] no ATF memory for init
[   13.886018] no ATF memory for init
[   13.897770] rk_gmac-dwmac fe010000.ethernet: no regulator found
[   14.302633] rk817-battery rk817-battery: Failed to find matching dt id
[   14.303254] rk817-charger rk817-charger: Failed to find matching dt id
[   14.314166] OF: graph: no port node found in /i2c@fe5c0000/fusb302@22
[   14.346316] arm-scmi firmware:scmi: Failed. SCMI protocol 17 not active.
[   14.353222] rksfc_base v1.1 2016-01-08
[   14.402548] mpp_rkvenc fdf40000.rkvenc: Failed to get leakage
[   14.403256] mpp_rkvenc fdf40000.rkvenc: failed to find power_model node
[   14.403375] mpp_rkvenc fdf40000.rkvenc: failed to initialize power model
[   14.403472] mpp_rkvenc fdf40000.rkvenc: failed to get dynamic-coefficient
[   14.405453] mpp_rkvdec2 fdf80200.rkvdec: shared_niu_a is not found!
[   14.405586] rkvdec2_init:1022: No niu aclk reset resource define
[   14.405677] mpp_rkvdec2 fdf80200.rkvdec: shared_niu_h is not found!
[   14.405766] rkvdec2_init:1025: No niu hclk reset resource define
[   14.406707] mpp_rkvdec2 fdf80200.rkvdec: failed to find power_model node
[   14.406815] mpp_rkvdec2 fdf80200.rkvdec: failed to initialize power model
[   14.406912] mpp_rkvdec2 fdf80200.rkvdec: failed to get dynamic-coefficient
[   14.410309] rockchip-dmc dmc: failed to get vop pn to msch rl
[   14.410704] rockchip-dmc dmc: could not find power_model node
[   14.977809] mmc1: delay init for 600 ms to enable UHS mode
```

Flashed on the onboard eMMC:
```
./rkdeveloptool db rk356x_spl_loader_ddr1056_v1.10.111.bin
./rkdeveloptool ef
./rkdeveloptool wl 0 radxa-zero3_debian_bullseye_cli_b3.img
```

Logs:
```
Starting kernel ...
 
[    9.620220] fiq_debugger fiq_debugger.0: IRQ fiq not found
[    9.620283] fiq_debugger fiq_debugger.0: IRQ wakeup not found
[    9.620317] fiq_debugger_probe: could not install nmi irq handler
[   11.095204] mpp-iep2 fdef0000.iep: allocate roi buffer failed
[   11.115423] arm-scmi firmware:scmi: Failed. SCMI protocol 22 not active.
[   11.138057] dwhdmi-rockchip fe0a0000.hdmi: IRQ index 1 not found
[   11.398239] no ATF memory for init
[   11.419250] no ATF memory for init
[   11.431232] rk_gmac-dwmac fe010000.ethernet: no regulator found
[   11.830715] rk817-battery rk817-battery: Failed to find matching dt id
[   11.831331] rk817-charger rk817-charger: Failed to find matching dt id
[   11.842223] OF: graph: no port node found in /i2c@fe5c0000/fusb302@22
[   11.874661] arm-scmi firmware:scmi: Failed. SCMI protocol 17 not active.
[   11.881973] rksfc_base v1.1 2016-01-08
[   11.924888] mpp_rkvenc fdf40000.rkvenc: Failed to get leakage
[   11.925562] mpp_rkvenc fdf40000.rkvenc: failed to find power_model node
[   11.925669] mpp_rkvenc fdf40000.rkvenc: failed to initialize power model
[   11.925766] mpp_rkvenc fdf40000.rkvenc: failed to get dynamic-coefficient
[   11.927641] mpp_rkvdec2 fdf80200.rkvdec: shared_niu_a is not found!
[   11.927769] rkvdec2_init:1022: No niu aclk reset resource define
[   11.927864] mpp_rkvdec2 fdf80200.rkvdec: shared_niu_h is not found!
[   11.927954] rkvdec2_init:1025: No niu hclk reset resource define
[   11.928879] mpp_rkvdec2 fdf80200.rkvdec: failed to find power_model node
[   11.928988] mpp_rkvdec2 fdf80200.rkvdec: failed to initialize power model
[   11.929086] mpp_rkvdec2 fdf80200.rkvdec: failed to get dynamic-coefficient
[   11.932530] rockchip-dmc dmc: failed to get vop pn to msch rl
[   11.936205] rockchip-dmc dmc: could not find power_model node
```

Please provide working images for Zero3W.

**Comment by RadxaYuntian** [comment]
See https://github.com/radxa-build/rock-3c/issues/5#issuecomment-1873584909

**Comment by zehnm** [comment]
@RadxaYuntian please don't close this issue, it's not fixed in b4 and other users will just fall into the same trap:
- The `radxa-zero3_debian_bullseye_cli_b4.img` also doesn't boot!
  - Same kernel logs as with the b3 image, so I don't repost them. 
  - Boot still hangs after message `rockchip-dmc dmc: could not find power_model node`
- The recommended "fix" in the referenced comment "to not use the CLI image and use the desktop variant instead" is not a solution.
  I would prefer to use CLI and have no use for a desktop, except for initially trying out a few things.

I suggest to update the release description on GitHub, that the Debian CLI version is currently broken until this is fixed.

**Comment by rohanrehman** [comment]
@zehnm Reopening issue to help paying customers.

**Comment by danielfaust** [comment]
Also, while at it, they could bump it up to Debian 12 (bookworm).

**Comment by RadxaYuntian** [comment]
Rockchip currently only released Debian 12 SDK for RK3588, so we will focus on that platform first. Once the SDK for RK356X is available we will start working on that as well.

**Comment by danielfaust** [comment]
It's now two months later. Could you please explain how the issue is progressing?

**Comment by RadxaYuntian** [comment]
Rockchip Linux 6.1 SDK for RK356X is not released yet.

**Comment by vinicentus** [comment]
Why not just release a new build with the fix where panfrost is disabled? While we wait for the debian bookworm and linux kernel 6.1 sdk, since that might still take a long time to arrive. That way it's at least easy for users to get started using this without having to build their own images or modyfying the existing ones.

**Comment by danielfaust** [comment]
@vinicentus For what it's worth, Radxa Zero 3W/3E is now supported by Armbian: https://www.armbian.com/radxa-zero-3/

**Comment by RadxaYuntian** [comment]
> Why not just release a new build with the fix where panfrost is disabled?

Because it has to go through the QA before it can be released, which is overloaded. We also do not officially support CLI variant partly due to lack of QA resources.

**Comment by Generalkidd** [comment]
Does the older debian_bullseye_cli_b2 boot or is every Debian CLI image broken? Also does Ubuntu CLI boot fine?

**Comment by Silence0314** [comment]
So is there a solution to this problem? I just buy a ZERO 3E and it seems that the xfce version of Debian that radxa provides has the same problem. I burned that b6 version of debian, and the log from serial just hung up after this:
[14.575241] rockchip-dmc dmc:  could not find power_model node

**Comment by RadxaYuntian** [comment]
@Silence0314 can you send a picture about your power supply's spec?

**Comment by Silence0314** [comment]
Sorry for the delay. Its a 5V3A power adapter. I have asked the customer service of the retailer all this week and after communication their techical support said that there maybe a problem with the board (since I have burned other systems like Ubuntu CLl and desktop, it also have that problem, which seems  it's not the system image's fault). I have shipped it back to the retailer  for repairment. BIG THANKS for your replys, you guys are really nice.
